examples/step-55/doc/intro.dox

```diff
@@ -0,0 +1,141 @@
+<h1>Introduction</h1>
+
+<h3> Stokes Problem </h3>
+
+The purpose of this tutorial is to create an efficient linear solver
+for the Stokes equation and compare it to alternative
+approaches. Using FGMRES with geometric multigrid as a precondtioner
+for the velocity block, we see that the linear solvers used in Step-22
+cannot keep up since multigrid is the only way to get $O(n)$ solve
+time. Using the Timer class, we collect some statistics to compare
+setup times, solve times, and number of iterations. We also compute
+errors to make sure that what we have implemented is correct.
+
+Let $u \in H_0^1 = \{ u \in H^1(\Omega), u|_{\partial \Omega} = 0 \}$
+and $p \in L_*^2 = \{ p_f \in L^2(\Omega), \int_\Omega p_f = 0
+\}$. The Stokes equations read as follows in non-dimensionalized form:
+
+@f{eqnarray*} - 2 \text{div} \frac {1}{2} \left[ (\nabla \textbf{u}) +
+(\nabla \textbf{u})^T\right] + \nabla p & =& f \\ - \nabla \cdot u &=&
+0 @f}
+
+Note that we are using the deformation tensor instead of $\Delta u$ (a
+detailed desription of the difference between the two can be found in
+Step-22, but in summary, the deformation tensor is more physical as
+well as more expensive).
+
+<h3> Linear solver and preconditioning issues </h3> The weak form of
+the discrete equations naturally leads to the following linear system
+for the nodal values of the velocity and pressure fields:
+@f{eqnarray*} 
+\left(\begin{array}{cc} A & B^T \\ B & 0
+\end{array}\right) \left(\begin{array}{c} U \\ P \end{array}\right) =
+\left(\begin{array}{c} F \\ 0 \end{array}\right), @f}
+
+Our goal is to compare several solver approaches.  In contrast to the
+way in which step-22 solves the Stokes equation, we instead attack the
+block system at once using a direct solver or FMGRES with an efficient
+preconditioner. The idea is as follows: if we find a block
+preconditioner $P$ such that the matrix @f{eqnarray*}
+\left(\begin{array}{cc} A & B^T \\ B & 0 \end{array}\right) P^{-1} @f}
+
+is simple, then an iterative solver with that preconditioner will
+converge in a few iterations. Notice that we are doing right
+preconditioning for this.  Using the Schur complement $S=BA^{-1}B^T$,
+we find that
+
+@f{eqnarray*}
+P^{-1} = \left(\begin{array}{cc} \hat{A} & B^T \\ 0 &
+ \hat{S} \end{array}\right)^{-1} @f}
+
+is a good choice. It is important to note that
+@f{eqnarray*} 
+P =
+\left(\begin{array}{cc} A^{-1} & 0 \\ 0 & I \end{array}\right)
+\left(\begin{array}{cc} I & B^T \\ 0 & -I \end{array}\right)
+\left(\begin{array}{cc} I & 0 \\ 0 & S^{-1} \end{array}\right).  @f}
+
+Since $P$ is aimed to be a preconditioner only, we shall use
+approximations to the inverse of the Schur complement $S$ and the
+matrix $A$.  Therefore, in the above equations, $-M_p=\hat{S} \approx
+S$, where $M_p$ is the pressure mass matrix and is solved by using CG
++ ILU, and $\hat{A}$ is an approximation of $A$ obtained by one of
+multiple methods: CG + ILU, just using ILU, CG + GMG (Geometric
```

Comment:
same here. 

I think it is a bit confusing to say that `\hat A` is an approximation to `A`. What it really is is that `\hat{A^{-1}}` is an approximation to `A^{-1}` where I've purposefully placed the hat onto the *inverse* of the matrix. In practice, this doesn't render particularly nicely, and using `\widetilde{A^{-1}}` is a better choice.